pkgs/applications/office/espanso/default.nix

```diff
@@ -1,69 +1,108 @@
 { lib
 , fetchFromGitHub
 , rustPlatform
+, cargo-make
 , pkg-config
 , extra-cmake-modules
+, dbus
 , libX11
 , libXi
 , libXtst
 , libnotify
+, libxkbcommon
 , openssl
 , xclip
 , xdotool
+, setxkbmap
+, wl-clipboard
+, wxGTK32
 , makeWrapper
 , stdenv
 , AppKit
 , Cocoa
 , Foundation
+, IOKit
+, waylandSupport ? stdenv.isLinux
+, x11Support ? stdenv.isLinux
+, testers
+, espanso
 }:
 
 rustPlatform.buildRustPackage rec {
   pname = "espanso";
-  version = "0.7.3";
+  version = "2.1.8";
 
   src = fetchFromGitHub {
-    owner = "federico-terzi";
+    owner = "espanso";
     repo = pname;
     rev = "v${version}";
-    sha256 = "1q47r43midkq9574gl8gdv3ylvrnbhdc39rrw4y4yk6jbdf5wwkm";
+    sha256 = "06wzrjdvssixgd9rnrv4cscbfiyvp5pjpnrih48r0ral3pj2hdg5";
   };
 
-  cargoSha256 = "0ba5skn5s6qh0blf6bvivzvqc2l8v488l9n3x98pmf6nygrikfdb";
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+    outputHashes = {
+      "yaml-rust-0.4.6" = "sha256-wXFy0/s4y6wB3UO19jsLwBdzMy7CGX4JoUt5V6cU7LU=";
+    };
+  };
 
   nativeBuildInputs = [
     extra-cmake-modules
     pkg-config
     makeWrapper
+    cargo-make
+    wxGTK32
```

Comment:
wxGTK32 should be buildINputs

Reply:
Strangely it fails otherwise.

Reply:
Hmm, that was suggested before and the build failed. I tried again to move this to `buildInputs` and indeed it fails again.

> wxWidgets is not installed, as `wx-config` cannot be executed

<details>

```
       > error: failed to run custom build command for `espanso-modulo v0.1.0 (/build/source/espanso-modulo)`
       >
       > Caused by:
       >   process didn't exit successfully: `/build/source/target/release/build/espanso-modulo-8653a240257a137b/build-script-build` (exit status: 101)
       >   --- stderr
       >   thread 'main' panicked at 'wxWidgets is not installed, as `wx-config` cannot be executed', espanso-modulo/build.rs:467:5
       >   note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
       > warning: build failed, waiting for other jobs to finish...
       For full logs, run 'nix log /nix/store/4ysxadjwjrmx68cyddxcc3kzi6ygd4ag-espanso-2.1.8.drv'.
```

</details>

This originates from https://github.com/espanso/espanso/blob/97f97a1906b0a54cb3ec20bb52ae6b08e609e9d6/espanso-modulo/build.rs#L450-L468

The problem seems to be that if I add wxGTK32 to buildInputs and remove it from nativeBuildInputs it won't show up in `PATH` and thus `wx-config` is not available during the build.

Having it in both does work, but I'm guessing that's even worse.

Is the problem that it is a build tool (like pkg-config), but also a library/dependency? I don't think I ran into such a problem before, but I'm also confused as I see that almost all other packages have it in `buildInputs`.

Reply:
> Having it in both does work, but I'm guessing that's even worse.

Because of the configure script that should be the correct solution and should also work when running with strictDeps = true which is used by cross compilation

Reply:
Thanks! Good to know. I've added wxGTK to buildInputs and nativeBuildInputs.